Automattic/woocommerce-payments
# Fixes Empty Alert When Refunding Authorization Charge

Fixes #741 

#### Changes proposed in this Pull Request

* Update to the `process_refund` logic to do a check to see if an order has not yet been charged (if it is still in authorization stage). If this is the case, instead of doing the refund, redirect the user to instead cancel the order via the 'Order actions' menu.

#### Testing instructions


* Navigate to `Payments -> Settings` and turn 'Manual Capture' **on**.
* As a customer, make a test order (it doesn't matter what is ordered).
* Navigate to `WooCommerce -> Orders` and click into the order that was created.
* Click the 'Refund' button, then 'Refund via WooCommerce'.
* On `trunk`, an empty alert should appear and an Order Note with the error message should be created. On this branch, an alert should pop up informing the user an authorization cannot be refunded, and directing them to the order actions menu should appear, and no note should be created.

-------------------
